designdocs/Capacity.md

```diff
@@ -0,0 +1,753 @@
+  UnStake {
+    /// The token account that staked tokens to the network.
+    account: T::AccountId,
+    /// The MSA that a token account targeted to receive Capacity.
+    msa_id: MessageSourceId,
+    /// An amount that was unstaked.
+    amount: BalanceOf<T>
+   },
+
+  /// The token amount that was previously unstaked and now withdrawn by unlocking. 
+  Withdrawn {
+    /// The token account that staked tokens to the network.
+    account: T::AccountId,
+    /// An amount that was withdrawn.
+    amount: BalanceOf<T>
+
+  }
+}
+```
+
+### **Storage**
+
+**Staking Storage**
+
+Storage for keeping records of staking accounting:
+
+```rust
+/// Storage for keeping a ledger of staked token amounts for accounts.
+#[pallet::storage]
+pub type StakingAccountLedger<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, StakingAccountDetails<T::Balance>>;
+```
+
+```rust
+/// Storage for keeping a record of how many tokens were targeted to an MSA.
+#[pallet::storage]
+pub type StakingTargetLedger<T: Config> = StorageDoubleMap<_, Blake2_128Concat, T::AccountId, Twox64Concat, MessageSourceId, StakingTargetDetails<T::Balance>>;
+```
+
+<!-- add maximiize and not maximize ( staking account ledger) -->
+
+```rust
+pub struct StakingTargetDetails<> {
+    stake: (capacity-rate, amount)
+
+}
+
+```
+
+The type used for storing information about staking details:
+
+```rust
+pub struct StakingAccountDetails<Balance, BlockNumber> {
+  /// The amount staker has actively at stacked (total - sum of unlocking).
+  pub active: Balance
+  /// The total amount staker has locked.
+  pub total: Balance,
+  /// Unstaked balances that will be made available.
+  pub unlocking: BoundedVec<UnlockChunk<BalanceOf<T>>, T::MaxUnlockingChunks>,
+}
+```
+
+```rust
+pub struct TargetDetails<Balance> {
+
+}
+```
+
+```rust
+pub struct UnlockChunk<Balance: HasCompact> {
+	/// Amount to be unlocked.
+	#[codec(compact)]
+	value: Balance,
+	/// Epoch number at which point funds are unlocked.
+	#[codec(compact)]
+	epoch: EpochIndex,
+}
+```
+
+
+### **Interfaces for Capacity-Pallet**
+
+### **Calls**
+
+**Set_epoch_period**
+
+```rust
+/// Sets the Epoch Period through governance.
+///
+/// ### Errors
+///
+/// - Returns Error::BadOrigin if sender is not root.
+pub fn set_epoch_period(origin: OriginFor<T>, blocks: MessageSourceId) {}
+```
+
+Acceptance Criteria are listed below but can evolve:
+
+1. Origin is Root.
+2. Sets the new Epoch-Period.
+3. New Epoch-Period begins at the Next Epoch's start.
+
+**Capacity Storage**
+
+Storage for the issuance of Capacity to Registered Providers:
+
+```rust
+/// Storage for an MSA's Capacity balance details.
+#[pallet::storage]
+pub type CapacityOf<T: Config> = StorageMap<_, Blake2_128Concat, MessageSourceId, CapacityDetails<T::Balance>>;
+```
+
+The type for storing Registered Provider Capacity balance:
+
+```rust
+pub struct CapacityDetails<Balance> {
+  /// The amount of Capacity used for an Epoch.
+  pub used: Balance,
```

Comment:
```suggestion
  pub remaining: Balance,
```

Reply:
New Capacity in Epoch 10

remaining: 0
total_available: 5
last_rep_epoch: 10

... epoch 11...

Start of Call:
remaining: 0
total_available: 5
last_rep_epoch: 10 <-- OLD

Update:
remaining: 5 - [Current Needs] = 4.5
total_available: 5
last_rep_epoch: 11

... epoch 12.....
.. epoch 13...

Start of Call:
remaining: 4.5
total_available: 5
last_rep_epoch: 11 <-- OLD

Update:
remaining: 5 - [Current Needs] = 4.2
total_available: 5
last_rep_epoch: 13

ANOTHER CALL IN 13

Start of Call:
remaining: 4.2
total_available: 5
last_rep_epoch: 13

Update:
Start of Call:
remaining: 4.2 - [Current Need] ONLY UPDATE THIS FIELD
total_available: 5
last_rep_epoch: 13 <--- CURRENT! 🎉